specs/stein/approved/amd-sev-libvirt-support.rst

```diff
@@ -0,0 +1,367 @@
+  states is not supported in either direction
+  <https://github.com/qemu/qemu/commit/8fa4466d77b44f4f58f3836601f31ca5e401485d>`_,
+  prevent nova from resizing an instance from a non-SEV-enabled flavor
+  to an SEV-enabled one, or vice-versa.  Alternatively, nova could
+  catch the error raised by QEMU, which would be propagated via
+  libvirt, and handle it appropriately.
+
+  Similarly, attempts to suspend / resume a domain should either be
+  prevented, or the error caught and handled.
+
+Alternatives
+------------
+
+#. The idea of initially using a custom trait could be skipped in
+   favour of immediately implementing the built-in trait
+   (``HW_CPU_x86_SEV`` or similar).  This would increase the amount of
+   effort required in order to reach a working prototype, but
+   conversely has the advantage of not requiring the merging of
+   unfinished proof-of-concept code into nova's ``master`` branch.
+
+#. Rather than using a single trait to both facilitate the matching
+   of instances requiring SEV with SEV-capable compute hosts *and*
+   indicate to nova's libvirt driver that SEV should be used when
+   booting ...
+
+   However this would create extra friction for the operators defining SEV-enabled
+   flavors,  it is hard to imagine
+
+#. Rather than traits, normal flavor extra specs could be used to
+   require the SEV feature; however it is understood that this
+   approach is `deprecated in favour of using traits
+   <https://docs.openstack.org/nova/latest/admin/configuration/schedulers.html#computecapabilitiesfilter>`_.
+
+
+Data model impact
+-----------------
+
+Changes which require modifications to the data model often have a wider impact
+on the system.  The community often has strong opinions on how the data model
+should be evolved, from both a functional and performance perspective. It is
+therefore important to capture and gain agreement as early as possible on any
+proposed changes to the data model.
+
+Questions which need to be addressed by this section include:
+
+* What new data objects and/or database schema changes is this going to
+  require?
+
+* What database migrations will accompany this change.
+
+* How will the initial set of new data objects be generated, for example if you
+  need to take into account existing instances, or modify other existing data
+  describe how that will work.
+
+REST API impact
+---------------
+
+None, although future work may require extending the REST API so that
+users can verify the hardware's attestation that the memory was
+encrypted correctly by the firmware.
+
+Security impact
+---------------
+
+There are no security impacts for the cloud deployment or cloud
+operator.  Tenants of the cloud will benefit from the ability to
+launch encrypted VMs.
+
+It is strongly recommended that `the OpenStack Security Group
+<openstack-security@lists.openstack.org>`_ is kept in the loop and
+invited to review each stage of work.
+
+Notifications impact
+--------------------
+
+None
```

Comment:
we might want to expose theinformation that the instance is running encrypted? a billing cloud provider might want to bill encrypted instances differently than unencrypted ones ("security surcharge")

Reply:
Good point - out of scope for this spec, but I'll mention the idea.